### PR TITLE
Jest snapshot testing for brewkit components 

### PR DESCRIPTION
**Pull request by kbulatovic**
## Description
Updated some deps for ts, jest bridge stuff.
Created a simple test cases for Alert component. 
Didn't cover the Material UI abstraction props, since they themselves cover them, so we rely on them to just work. 
IMO our test cases should just cover custom changes on the components.

## Additional info
Added jest plugin for storybook UI, so we can monitor test snapshots easier, covered few more components.

Regarding the naming convention for test files, needed to revert to [componentName].test.tsx, since the storybook jest addon requires unique naming convention for each component, so the generic test.tsx as a filename will resolve the first test case to be shown globally on every storybook component. It is the way the source code is written for the jest storybook plugin.

TODO
Maybe split the storybook component segments into smaller parts so we can write particular tests for example use cases ie - test for onChange prop, if the example demonstrates onChange prop usage?

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by ItsEricWoodward — APPROVED** [review]

**Review by emdrayallday — CHANGES_REQUESTED** [review]
I see you removed a lot of double spacing, this is intentional. We want to separate sections with double spacing and content within a section by single space, just to make things more readable and deliberate.

```
import React from 'react';
import MuiAlert from '@material-ui/lab/Alert';
// 2 empty lines
// 
const Alert = ({
    ...props
}) => {
// 2 empty lines
// 
    const [state, setState] = React.useState('state');
    const [otherState, setOtherState] = React.useState('otherState');
// 2 empty lines
// 
   const handlerFunction = () => {
         // function stuff
    }
    // 1 empty line b/c these are similar but need to be separated to make it more clear
   const otherFunction = () => {
        // other function stuff
    }
// 2 empty lines
// 
    return (
         JSX
    );
}
// 2 empty lines
// 
export default Alert;

File: src/components/CircularProgress/stories.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import CircularProgress from '@components/CircularProgress';
 export default {
     component: CircularProgress,
     parameters: {
-        jest: ['CircularProgress.test']
+        jest: ['CirularProgress.test']
```
Comment (emdrayallday) [inline src/components/CircularProgress/stories.tsx:9]: typo

**Comment by lawren** [comment]
@kbulatovic, I updated Storybook which changed some of the story formatting. That's most likely where the conflicts are coming from there. You should just have to re-add the jest config to the story parameters.

**Comment by ItsEricWoodward** [comment]
I'm closing this PR because of its age and size. I like where this is going, and I encourage you to resubmit it in smaller chunks, and with the conflicts resolved. Thanks!

**Pull request closed without merging**